File: docs/components_page/components/form/feedback.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

email_input <- htmlDiv(
  list(
    dbcLabel("Email"),
    dbcInput(id = "email-input", type = "email", value = ""),
    dbcFormText("We only accept gmail..."),
    dbcFormFeedback(
      "That looks like a gmail address :-)",
      type = "valid"
    ),
    dbcFormFeedback(
      "Sorry, we only accept gmail for some reason...",
      type = "invalid"
    )
  )
)


# --- Callbacks --- #
app$callback(
  list(
    output("email-input", "valid"),
    output("email-input", "invalid")
  ),
  list(
    input("email-input", "value")
  ),
  function(text) {
    if (text != "") {
      is_gmail <- endsWith(text, "@gmail.com")
      return(list(is_gmail, !is_gmail))
    }
    return(list(FALSE, FALSE))
  }
)

File: docs/components_page/components/form/feedback.jl
using DashBootstrapComponents

email_input = html_div([
    dbc_label("Email"),
    dbc_input(id = "email-input", type = "email", value = ""),
    dbc_formtext("We only accept gmail..."),
    dbc_formfeedback("That looks like a gmail address :-)", type = "valid"),
    dbc_formfeedback("Sorry, we only accept gmail for some reason...", type = "invalid"),
]);


callback!(app, Output("email-input", "invalid"), Input("email-input", "value")) do text
    if length(text) > 0
        is_gmail = endswith(text, "@gmail.com")
        return !is_gmail
    end
    return false
end;

callback!(app, Output("email-input", "valid"), Input("email-input", "value")) do text
    if length(text) > 0
        is_gmail = endswith(text, "@gmail.com")
        return is_gmail
    end
    return false
end;
